src/test/java/org/eclipse/tracecompass/trace_event_logger/SnapshotTest.java

```diff
@@ -0,0 +1,242 @@
+    @Before
+    public void before() throws SecurityException, IOException {
+        fLogger = Logger.getAnonymousLogger();
+        fStreamHandler = new SnapshotHandler(0.5);
+        for (Handler handler : fLogger.getHandlers()) {
+            fLogger.removeHandler(handler);
+        }
+        fLogger.addHandler(fStreamHandler);
+        fLogger.setLevel(Level.ALL);
+        fStreamHandler.setLevel(Level.ALL);
+    }
+
+    /**
+     * Test something too fast for the snapshot
+     */
+    @Test
+    public void fastTest() {
+        Logger logger = this.fLogger;
+        assertNotNull(logger);
+        logger.log(new LogUtils.TraceEventLogRecord(Level.INFO, () -> "", 0L, 'a', "Bla")); //$NON-NLS-1$ //$NON-NLS-2$
+        logger.log(new LogUtils.TraceEventLogRecord(Level.INFO, () -> "", 0L, 'B', "Bla")); //$NON-NLS-1$ //$NON-NLS-2$
+        logger.log(new LogUtils.TraceEventLogRecord(Level.INFO, () -> "", 0L, 'E', "Bla")); //$NON-NLS-1$ //$NON-NLS-2$
+        fStreamHandler.flush();
+    }
+
+    /**
+     * Test malformed events
+     */
+    @Test
+    public void badTest() {
+        Logger logger = this.fLogger;
+        assertNotNull(logger);
+        logger.log(new LogUtils.TraceEventLogRecord(Level.INFO, () -> "", 0L, 'a')); //$NON-NLS-1$
+        logger.log(new LogUtils.TraceEventLogRecord(Level.INFO, () -> "", 0L, 'B')); //$NON-NLS-1$
+        logger.log(new LogUtils.TraceEventLogRecord(Level.INFO, () -> "", 0L, 'E')); //$NON-NLS-1$
+        fStreamHandler.flush();
+    }
+
+    /**
+     * Test an actual snapshot
+     *
+     * @throws InterruptedException
+     *             won't happen
+     * @throws IOException
+     *             won't happen
+     */
+    @Test
+    public void slowTest() throws InterruptedException, IOException {
+        Logger logger = this.fLogger;
+        assertNotNull(logger);
+        logger.log(new LogUtils.TraceEventLogRecord(Level.INFO, () -> "\"a\"", 10000000000L, 'a', "Bla")); //$NON-NLS-1$ //$NON-NLS-2$
+        logger.log(new LogUtils.TraceEventLogRecord(Level.INFO, () -> "\"b\"", 20000000000L, 'B', "Bla")); //$NON-NLS-1$ //$NON-NLS-2$
+        logger.log(new LogUtils.TraceEventLogRecord(Level.INFO, () -> "\"c\"", 30000000000L, 'c', "Bla")); //$NON-NLS-1$ //$NON-NLS-2$
+        logger.log(new LogUtils.TraceEventLogRecord(Level.INFO, () -> "\"d\"", 40000000000L, 'd', "Bla")); //$NON-NLS-1$ //$NON-NLS-2$
+        logger.log(new LogUtils.TraceEventLogRecord(Level.INFO, () -> "\"e\"", 50000000000L, 'e', "Bla")); //$NON-NLS-1$ //$NON-NLS-2$
+        logger.log(new LogUtils.TraceEventLogRecord(Level.INFO, () -> "\"f\"", 60000000000L, 'f', "Bla")); //$NON-NLS-1$ //$NON-NLS-2$
+        logger.log(new LogUtils.TraceEventLogRecord(Level.INFO, () -> "\"g\"", 70000000000L, 'E', "Bla")); //$NON-NLS-1$ //$NON-NLS-2$
+        Thread.sleep(1000);
+        fStreamHandler.flush();
+        File input = new File("request-10000000.json"); //$NON-NLS-1$
+        try (FileReader fr = new FileReader(input)) {
```

Comment:
I get a failure in the test when running from maven. Also, sometimes when running in Eclipse IDE. It seems that the file is not being written in time before it is read here:
```
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   SnapshotTest.slowTest:132 » FileNotFound request-10000000.json (No such file o...
[INFO] 
[ERROR] Tests run: 39, Failures: 0, Errors: 1, Skipped: 0
```

Reply:
Now it works